src/debug-generator/debug-generator.c

```diff
@@ -239,7 +242,7 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
                 if (r < 0)
                         return log_oom();
 
-                install_debug_shell_dropin();
+                RET_GATHER(r, install_debug_shell_dropin());
```

Comment:
What's the point of `RET_GATHER` here? We fail just before if `r < 0`

Reply:
So that failure from proc_cmdline_parse is not discarded? I changed `strv_extend` above to not use `r` now.